components/Wizard/OCIAuthStep.js

```diff
@@ -0,0 +1,174 @@
+import React from "react";
+import { Button, Form, Popover, Text, TextInput, FileUpload } from "@patternfly/react-core";
+import { OutlinedQuestionCircleIcon } from "@patternfly/react-icons";
+import { FormattedMessage, defineMessages, injectIntl, intlShape } from "react-intl";
+import PropTypes from "prop-types";
+
+const ariaLabels = defineMessages({
+  user: {
+    id: "user-help",
+    defaultMessage: "User OCID",
+  },
+  private_key: {
+    id: "key-help",
+    defaultMessage: "Private key",
+  },
+  fingerprint: {
+    id: "fingerprint-help",
+    defaultMessage: "API private key's fingerprint",
+  },
+});
+
+class OCIAuthStep extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = { value: "", filename: "", isLoading: false };
+    this.handleFileInputChange = (event, file) => this.setState({ filename: file.name });
```

Comment:
I would recommend also adding a parameter to the props so that the filename is saved in the props and loaded from there. Currently, if a user changes to a different step and then comes back there is nothing showing they already uploaded a file.

Reply:
I may be missing something here - setting the state of `filename` here isn't not enough?

Reply:
So each step only renders when we are on the step. Changing steps will cause the step to unmount from the DOM. State variables will then be cleared. The whole wizard, however, does not unmount until the whole process is completed. So, any state stored int the CreateImageUpload component will not be cleared when changing steps but each step's state will be cleared. I will generally store temporary UI focused state (i.e. `isLoading`) in subcomponents and will store more important, longer-term data in the parent components or the redux store.

Reply:
done.